etc/adminhtml/system.xml

```diff
@@ -663,6 +663,57 @@
                         </depends>
                     </field>
                 </group>
+                <group id="omise_cc_googlepay" translate="label" sortOrder="505" showInDefault="1" showInWebsite="1" showInStore="1">
+                    <label>Enable Google Pay Payment</label>
+                    <comment>Enable customers to pay using credit/debit cards connected to their Google account. </comment>
+                    <field id="active" translate="label comment" type="select" sortOrder="506" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Enable/Disable</label>
+                        <comment>Enable Google Pay payment method.</comment>
+                        <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                        <config_path>payment/omise_cc_googlepay/active</config_path>
+                    </field>
+                    <field id="merchant_id" translate="label" type="text" sortOrder="507" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Google Pay Merchant ID</label>
+                        <comment><![CDATA[The merchant ID will be available after registering with the <a href="https://pay.google.com/business/console">Google Pay Business Console</a>.]]></comment>
+                        <config_path>payment/omise_cc_googlepay/merchant_id</config_path>
+                    </field>
+                    <field id="request_billing_address" translate="label comment" type="select" sortOrder="508" showInDefault="1" showInWebsite="1" showInStore="1">
+                        <label>Request Billing Address</label>
+                        <comment>Request customer's name and billing address from their Google Account upon checkout.</comment>
+                        <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                        <config_path>payment/omise_cc_googlepay/request_billing_address</config_path>
+                    </field>
+                    <field id="request_phone_number" translate="label comment" type="select" sortOrder="509" showInDefault="1" showInWebsite="1" showInStore="1">
```

Comment:
we will only fetch the phone number if billing address is also enabled right? is there any way to dynamically disable this if billing is disabled? or maybe a good idea to add in the description about this

Reply:
Nice idea! Added fix here [e5ea073](https://github.com/omise/omise-magento/pull/351/commits/e5ea073bc006540b7ad229e8dd19304279e807c5) and now it's not visible if billing address is not requested.

![Screen Shot 2565-05-27 at 17 15 10](https://user-images.githubusercontent.com/61586978/170681307-9285934e-7f83-470d-9f49-9c15b79e6f90.png)